lib/sync.js

```diff
@@ -394,7 +403,7 @@ SyncApi.prototype._sync = function(syncOptions, attempt) {
         since: syncToken || undefined // do not send 'null'
     };
 
-    if (attempt > 1) {
+    if (self._syncConnectionLost) {
```

Comment:
I believe so, since we only actually consider the connection back up once the sync comes back rather than when /versions returns. Possibly this is unnecessary but it feels more true to only emit the SYNCING event after we've actually got a new sync response.